bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -127,35 +127,41 @@ public GarbageCollectorThread(ServerConfiguration conf, LedgerManager ledgerMana
                 Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("GarbageCollectorThread")));
     }
 
+    private EntryLogMetadataMap createEntryLogMetadataMap(boolean gcPersistentEntrylogMetadataMapEnabled) {
+        final String baseDir = this.entryLogger.getLedgerDirsManager().getAllLedgerDirs().get(0).toString();
```

Comment:
what happens if you have more than one "ledgerDir" ? should this directory be configurable ?

maybe you can move this line inside the *if* block at line 133

Reply:
> what happens if you have more than one "ledgerDir" ? should this directory be configurable ?

actually, that doesn't matter because here we use it to just get the base dir path of ledger so, we can create dedicated dir to store metadata and that is independent of multi-ledger dir usecase. also we do similar at [multiple places](https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java#L2869) to get base-dir path for the entry-logger and stores rocksDB dir under it.
However, we can also add config to get dir path and fallback to this logic if config is not provided. I will add it.